### ISSUE TITLE
Failure to install distribute/setuptools into virtualenv from a 'fresh' win32 python 2.7 build because of wrong directory structure

### ISSUE THREAD

**Issue by EBNull** [pypa/virtualenv#139]
On windows, to freshly build Python 2.7:

```
wget http://www.python.org/ftp/python/2.7.2/Python-2.7.2.tar.bz2
tar -xf Python-2.7.2
//Make sure you're in the VS2008 build environ
cd Python-2.7.2\PCBuild
call build.bat
call build.bat -d
```

At this point, the *.pyd extension modules live next to python.exe (for example, _socket.pyd).

After installing distribute and virtualenv from this python.exe and then trying to create a virtualenv, I get this traceback:

```
New python executable in env\Scripts\python.exe
Installing distribute..............
  Complete output from command Z:\pyhack\env\Scripts\python.exe -c "#!python
\"\"\"Bootstra... main(sys.argv[1:])
" --always-copy -U distribute:
  Traceback (most recent call last):
  File "<string>", line 485, in <module>
  File "<string>", line 480, in main
  File "<string>", line 185, in download_setuptools
  File "Z:\pyhack\python\Lib\urllib2.py", line 94, in <module>
    import httplib
  File "Z:\pyhack\python\Lib\httplib.py", line 71, in <module>
    import socket
  File "Z:\pyhack\python\Lib\socket.py", line 47, in <module>
    import _socket
ImportError: No module named _socket
----------------------------------------
...Installing distribute...done.
Traceback (most recent call last):
  File "Z:\pyhack\python\scripts\virtualenv-script.py", line 9, in <module>
    load_entry_point('virtualenv==1.6.1', 'console_scripts', 'virtualenv')()
  File "Z:\pyhack\python\lib\site-packages\virtualenv.py", line 795, in main
    never_download=options.never_download)
  File "Z:\pyhack\python\lib\site-packages\virtualenv.py", line 892, in create_environment
    search_dirs=search_dirs, never_download=never_download)
  File "Z:\pyhack\python\lib\site-packages\virtualenv.py", line 593, in install_distribute
    search_dirs=search_dirs, never_download=never_download)
  File "Z:\pyhack\python\lib\site-packages\virtualenv.py", line 562, in _install_req
   cwd=cwd)
  File "Z:\pyhack\python\lib\site-packages\virtualenv.py", line 863, in call_subprocess
    % (cmd_desc, proc.returncode))
OSError: Command Z:\pyhack\env\Scripts\python.exe -c "#!python
\"\"\"Bootstra... main(sys.argv[1:])
" --always-copy -U distribute failed with error code 1
```

Apparently, the creation of the virtualenv went OK, but it could not install distribute because it distribute's install script (eventually) can't import _socket.

Basically, if we're on win32, python assumes a standard layout for the existence of the *pyd extensions. As this is a python source build, this is not the case - the *pyd files live next to python.exe. The resulting interpreter does not place it's old path on the resulting sys,path, which eventually leads to an ImportError when trying to load a .pyd
